src/onelogin/saml2/utils.py

```diff
@@ -897,10 +909,8 @@ def validate_node_sign(signature_node, elem, cert=None, fingerprint=None, finger
             dsig_ctx.key = xmlsec.Key.from_memory(cert, xmlsec.KeyFormat.CERT_PEM, None)
 
         dsig_ctx.set_enabled_key_data([xmlsec.KeyData.X509])
-        try:
-            dsig_ctx.verify(signature_node)
-        except Exception:
-            raise Exception('Signature validation failed. SAML Response rejected')
+        dsig_ctx.verify(signature_node)
```

Comment:
I think the `dsig_ctx.verify(signature_node)` code should be inside try except block and then the raised exception should be converted to `OneLogin_Saml2_Error` exception. This way the user of this library doesn't need to consider different exception types raised by the xmlsec library, and catching `OneLogin_Saml2_Error` exceptions will cover invalid signature exceptions as well.